pkg/cli/set/routebackends.go

```diff
@@ -93,7 +92,7 @@ type BackendsOptions struct {
 
 func NewBackendsOptions(streams genericclioptions.IOStreams) *BackendsOptions {
```

Comment:
I have not, but can you please test `oc set route-backends` properly? It seems we are using `scheme.DefaultJSONEncoder()`  in `CalculatePatchesExternal` which is used here and it does not have route in the schema as it uses the default one..

Reply:
I have verified that `oc set route-backends` works without any problem with this patch. I think, it works with `scheme.DefaultJSONEncoder()` because default json encoder has a fallback mechanism for unregistered resources https://github.com/openshift/oc/blob/95512488a135d12d17cc271ff5fe9d57baa10baf/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go#L489

Reply:
hm, even if it works, if I look at the implementation in  
https://github.com/openshift/oc/blob/795bf1a6260847ecfc612da2ab11ea2d6e07da16/vendor/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go#L217

(hopefully the correct file) I can see bunch of logic for the recognized types (eg. version conversion), I do not fully understand the implications that might occur in the future ,but think it would be safer to either pass the scheme or the encoder to `CalculatePatchesExternal` so each caller can use their scheme

Reply:
I agree that safest would be to pass the scheme into CalculatePatchesExternal, I'll update with that.